addons/website_sale_stock/controllers/main.py

```diff
@@ -30,3 +32,38 @@ def shop_payment_transaction(self, *args, **kwargs):
         if values:
             raise ValidationError('. '.join(values) + '.')
         return super().shop_payment_transaction(*args, **kwargs)
+
+
+class WebsiteSale(website_sale_controller.WebsiteSale):
+    @http.route(['/shop/stock_email_notification'], type="http", auth="public", website=True, sitemap=False)
+    def stock_email_notification(self, email, product_id):
+        print(email, product_id)
+
+        if not email or not product_id:
+            return request.not_found()
+
+        website = request.env['website'].get_current_website()
+        pricelist = website.pricelist_id
+        product = request.env['product.product'].browse(product_id)
+
+        price = product._get_combination_info_variant(pricelist=website.pricelist_id, )['price']
+
+        wishlist_model = request.env['product.wishlist']
+        if request.website.is_public_user():
+            wishlist_model = wishlist_model.sudo()
+            partner_ids = request.env['res.partner'].sudo()._mail_find_partner_from_emails([email],
+                                                                                           force_create=True)
+            partner_id = partner_ids[0]
+        else:
+            partner_id = request.env.user.partner_id
+
+        wish = wishlist_model._add_to_wishlist(
+            pricelist.id,
+            pricelist.currency_id.id,
+            request.website.id,
+            price,
+            product_id,
+            partner_id
+        )
```

Comment:
To check with the PO: this will allow a public user to register an existing user to products. We might need to check that the partner/mail isn't linked to an existing user as well.